woocommerce/woocommerce
# Add JS lint scripts for woocommerce plugin

### Changes proposed in this Pull Request:





Closes #45826 .

Add lint scripts for JS files in woocommerce plugin project and include the script for changed files in the lint job running in CI. 

The e2e tests have many lint errors, but fixing them is not part of this PR. That will be addressed in future dedicated PRs or whenever those files will be touched, because with this PR fixing them will become mandatory to be able to merge.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

- Check the lint job `Lint - @woocommerce/plugin-woocommerce` in CI
	- Tested with a changed file with eslint errors: [some existing errors here](https://github.com/woocommerce/woocommerce/actions/runs/8552869786/job/23435001730#step:4:41) and [focused test error](https://github.com/woocommerce/woocommerce/actions/runs/8553057779/job/23435521818?pr=46214#step:4:29) - lint job failed as expected
	- Tested with a changed file with eslint warnings: [here](https://github.com/woocommerce/woocommerce/actions/runs/8553288141/job/23436450672#step:4:30) - lint job passed
	- No errors
- Locally, run the scripts added in package.json. Eslint errors and warning are expected.
